test/Microsoft.AspNet.Mvc.Razor.Test/Precompilation/RazorErrorExtensionsTest.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Razor.Parser.SyntaxTree;
+using Microsoft.AspNet.Razor.Text;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.Razor.Test
+{
+    public class RazorErrorExtensionsTest
+    {
+        public static TheoryData ToDiagnostic_DoesNotThrowWhenRazorErrorLocationIsZeroOrUndefinedData
+        {
+            get
+            {
+                return new TheoryData<SourceLocation, int>
+                {
+                    { SourceLocation.Undefined, -1 },
+                    { SourceLocation.Undefined, 0 },
+                    { SourceLocation.Zero, -1 },
+                    { SourceLocation.Zero, 0 },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ToDiagnostic_DoesNotThrowWhenRazorErrorLocationIsZeroOrUndefinedData))]
+        public void ToDiagnostic_DoesNotThrowWhenRazorErrorLocationIsZeroOrUndefined(
```

Comment:
nit: `DoesNotThrow` is as meaningless as the assertion that got pulled out of Xunit.  suggest `Succeeds`